### PR TITLE
NAS-105344 / 11.3 / Nas 105344

### PR DESCRIPTION
**Pull request by lola27**

### CODE REVIEW

**Comment by lola27** [comment]
@freenas/docs-team For the wizard, the readonly field is a checkbox, if checked, then set readonly to be SET, otherwise set it to be IGNORE. could you please provide tooltip for it? thanks

File: src/app/pages/task-calendar/replication/replication-form/replication-form.component.ts [hunk 1]
```diff
@@ -822,6 +841,29 @@ export class ReplicationFormComponent {
...
+
+        if (entityForm.pk === undefined) {
+            readonlyCtrl.setValue(isFreenas ? 'SET' : 'REQUIRE');
+        }
+
+        this.entityForm.formGroup.controls['transport'].valueChanges.subscribe(
+            (res) => {
+                for (const option of readonlyField.options) {
+                    if (res === 'LEGACY') {
+                        option['disable'] = isFreenas ? (option.value === 'SET' ? false : true) : (option.value === 'IGNORE' ? false : true);
```
Comment (themylogin) [inline src/app/pages/task-calendar/replication/replication-form/replication-form.component.ts]: It's `REQUIRE` for TrueNAS

File: src/app/pages/task-calendar/replication/replication-form/replication-form.component.ts [hunk 2]
```diff
@@ -822,6 +841,29 @@ export class ReplicationFormComponent {
...
+                    } else {
+                        option['disable'] = false;
+                    }
+                }
+                if (res === 'LEGACY' && ((isFreenas && readonlyCtrl.value !== 'SET') || (!isFreenas && readonlyCtrl.value !== 'IGNORE'))) {
+                   readonlyCtrl.setValue(isFreenas ? 'SET' : 'IGNORE');
```
Comment (themylogin) [inline src/app/pages/task-calendar/replication/replication-form/replication-form.component.ts]: Same here

File: src/app/helptext/task-calendar/replication/replication-wizard.ts [hunk 3]
```diff
@@ -87,6 +87,9 @@ export default {
  <i>Destination Dataset</i>. Select a preset schedule or choose \
  <i>Custom</i> to use the advanced scheduler.'),
 
+    readonly_placeholder: T('Destination dataset read-only'),
```
Comment (themylogin) [inline src/app/helptext/task-calendar/replication/replication-wizard.ts]: "Set destination dataset to be read-only"?

File: src/app/helptext/task-calendar/replication/replication-wizard.ts [hunk 4]
```diff
@@ -87,6 +87,9 @@ export default {
  <i>Destination Dataset</i>. Select a preset schedule or choose \
  <i>Custom</i> to use the advanced scheduler.'),
 
+    readonly_placeholder: T('Set Destination Dataset Read-only'),
```
Comment (Mrt134) [inline src/app/helptext/task-calendar/replication/replication-wizard.ts]:
```suggestion
    readonly_placeholder: T('Destination Dataset Read-only Policy'),
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 5]
```diff
@@ -173,6 +173,11 @@ export default {
  add the <b>Schedule</b> values to the \
  <b>Snapshot Replication Schedule</b>.'),
 
+    readonly_placeholder: T('Set Destination Dataset Read-only'),
```
Comment (Mrt134) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
    readonly_placeholder: T('Destination Dataset Read-only Policy'),
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 6]
```diff
@@ -173,6 +173,11 @@ export default {
...
+    readonly_tooltip: T('<b>SET</b> will set all destination datasets to readonly=on after finishing the replication <br>\
+ <b>REQUIRE</b> will require all existing destination datasets to have readonly=on property <br>\
+ <b>IGNORE</b> will avoid this kind of behavior'),
```
Comment (Mrt134) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
    readonly_tooltip: T('<b>SET</b> changes all destination datasets to <code>readonly=on</code> after finishing the replication. <br>\
 <b>REQUIRE</b> stops replication unless all existing destination datasets have the property <code>readonly=on</code>. <br>\
 <b>IGNORE</b> disables checking the <code>readonly</code> property during replication.'),
```

File: src/app/helptext/task-calendar/replication/replication-wizard.ts [hunk 7]
```diff
@@ -87,6 +87,9 @@ export default {
...
+    readonly_tooltip: T(''),
```
Comment (Mrt134) [inline src/app/helptext/task-calendar/replication/replication-wizard.ts]: Suggested help text:

Setting this option changes the destination dataset to be read-only. To continue using the default or existing dataset read permissions, leave this option unset.

**Pull request merged**